### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'output_dir'

### ISSUE THREAD

**Issue by ali0une** [jhj0517/sd-webui-AdvancedLivePortrait#4]
**Which OS are you using?**
 - OS: Debian 12
 - Ui: Forge version: [f2.0.1v1.10.1-previous-613-gf10214e3](https://github.com/lllyasviel/stable-diffusion-webui-forge/commit/f10214e358e730b495ce7bfb0edc4846caecf1b2)
 - python: 3.10.6
 - torch: 2.4.1+cu121
 - xformers: N/A
 - gradio: 4.40.0

i do have an Ad folder in /whatever/stable-diffusion-webui-forge/outputs/
i can see generated images and exp_data and temp folders in it.

But if i click the blue folder at bottom of the AdvancedLivePortrait tab : 

```
Traceback (most recent call last):
  File "/whatever/stable-diffusion-webui-forge/venv/lib/python3.10/site-packages/gradio/queueing.py", line 536, in process_events
    response = await route_utils.call_process_api(
  File "/whatever/stable-diffusion-webui-forge/venv/lib/python3.10/site-packages/gradio/route_utils.py", line 285, in call_process_api
    output = await app.get_blocks().process_api(
  File "/whatever/stable-diffusion-webui-forge/venv/lib/python3.10/site-packages/gradio/blocks.py", line 1923, in process_api
    result = await self.call_function(
  File "/whatever/stable-diffusion-webui-forge/venv/lib/python3.10/site-packages/gradio/blocks.py", line 1508, in call_function
    prediction = await anyio.to_thread.run_sync(  # type: ignore
  File "/whatever/stable-diffusion-webui-forge/venv/lib/python3.10/site-packages/anyio/to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/whatever/stable-diffusion-webui-forge/venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
  File "/whatever/stable-diffusion-webui-forge/venv/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "/whatever/stable-diffusion-webui-forge/venv/lib/python3.10/site-packages/gradio/utils.py", line 818, in wrapper
    response = f(*args, **kwargs)
  File "/whatever/stable-diffusion-webui-forge/extensions/sd-webui-AdvancedLivePortrait/scripts/advanced_live_portrait.py", line 84, in <lambda>
    fn=lambda: self.open_folder(self.args.output_dir), inputs=None, outputs=None
AttributeError: 'NoneType' object has no attribute 'output_dir'
```

P.S : wonderful extension, keep up the good work ...

**Comment by jhj0517** [comment]
Hi, thanks for reporting.
This is fixed in #5 and should work fine now.

Feel free to re open!

**Comment by ali0une** [comment]
Hi there!

On a Linux box i have the following error when i click the blue folder icon:
`sh: 1: start: not found`

The `os.system(f"start {folder_path}")` command in [advanced_live_portrait.py:98](https://github.com/jhj0517/sd-webui-AdvancedLivePortrait/blob/68bdcea4c1d50ab6d8ce7a298a36f6fdf3a9d87c/scripts/advanced_live_portrait.py#L98) is specific to Windows.

The [open_folder](https://github.com/jhj0517/sd-webui-AdvancedLivePortrait/blob/68bdcea4c1d50ab6d8ce7a298a36f6fdf3a9d87c/scripts/advanced_live_portrait.py#L94) function can be fixed like this to support also Linux and Mac :
```
import os
import subprocess
import sys

    @staticmethod
    def open_folder(folder_path: str):
        if not os.path.exists(folder_path):
            os.makedirs(folder_path, exist_ok=True)
            print(f"The directory path {folder_path} has newly created.")
        
        if os.name == 'nt':  # Windows
            os.startfile(folder_path)
        elif os.name == 'posix':  # macOS and Linux
            if sys.platform == 'darwin':  # macOS
                subprocess.Popen(['open', folder_path])
            elif sys.platform.startswith('linux'):  # Linux
                subprocess.Popen(['xdg-open', folder_path])
        else:
            print("Unsupported operating system")
```
